assets/js/googlesitekit/data/create-settings-store.js

```diff
@@ -0,0 +1,392 @@
+
+	const actions = {
+		/**
+		 * Sets settings for the given values.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param {Object} values Settings with their values to set.
+		 * @return {Object} Redux-style action.
+		 */
+		setSettings( values ) {
+			invariant( values, 'values is required.' );
+
+			return {
+				payload: { values },
+				type: SET_SETTINGS,
+			};
+		},
+
+		/**
+		 * Dispatches an action that creates an HTTP request to the settings endpoint.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 *
+		 * @return {Object} Redux-style action.
+		 */
+		fetchSettings() {
+			return {
+				payload: {},
+				type: FETCH_SETTINGS,
+			};
+		},
+
+		/**
+		 * Stores settings received from the REST API.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 *
+		 * @param {Array} values Settings with their values from the API.
+		 * @return {Object} Redux-style action.
+		 */
+		receiveSettings( values ) {
+			invariant( values, 'values is required.' );
+
+			return {
+				payload: { values },
+				type: RECEIVE_SETTINGS,
+			};
+		},
+
+		/**
+		 * Dispatches an action signifying the `fetchSettings` side-effect failed.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 *
+		 * @return {Object} Redux-style action.
+		 */
+		receiveSettingsFailed() {
+			return {
+				payload: {},
+				type: RECEIVE_SETTINGS_FAILED,
+			};
+		},
+
+		/**
+		 * Saves all current settings to the server.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @return {Object} Redux-style action.
+		 */
+		*saveSettings() {
+			try {
+				const savedValues = yield actions.fetchSaveSettings();
+				return actions.receiveSaveSettings( savedValues );
+			} catch ( err ) {
+				// TODO: Implement an error handler store or some kind of centralized
+				// place for error dispatch...
+				return actions.receiveResetFailed();
+			}
+		},
+
+		/**
+		 * Dispatches an action that creates an HTTP request to save settings.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 *
+		 * @return {Object} Redux-style action.
+		 */
+		fetchSaveSettings() {
+			return {
+				payload: {},
+				type: FETCH_SAVE_SETTINGS,
+			};
+		},
+
+		/**
+		 * Dispatches that settings were saved via the REST API.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 *
+		 * @param {Array} values Settings with their values from the API.
+		 * @return {Object} Redux-style action.
+		 */
+		receiveSaveSettings( values ) {
+			invariant( values, 'values is required.' );
+
+			return {
+				payload: {},
+				type: RECEIVE_SAVE_SETTINGS,
+			};
+		},
+
+		/**
+		 * Dispatches an action signifying the `fetchSaveSettings` side-effect failed.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 *
+		 * @return {Object} Redux-style action.
+		 */
+		receiveSaveSettingsFailed() {
+			return {
+				payload: {},
+				type: RECEIVE_SAVE_SETTINGS_FAILED,
+			};
+		},
+	};
+
+	const controls = {
+		[ FETCH_SETTINGS ]: () => {
+			return API.get( type, identifier, datapoint );
+		},
+		// TODO: How do we get the current values to be available here?
```

Comment:
Arguments passed to actions are also passed to their relevant controls: https://developer.wordpress.org/block-editor/packages/packages-data/#registering-a-store has an example of this.

So any settings to be `POST`ed to the API should be provided to the action creator and will be available to the control. I left a comment about for how that would work 🙂

Reply:
Implemented this, alongside some other changes to make that possible (`storeName` and `registry` awareness of these functions).